lua/Dockerfile

```
@@ -1,7 +1,16 @@
-FROM esolang/lua:2.4.0
+FROM mileschou/lua:5.4
```

```suggestion
FROM nickblah/lua:5.4-luarocks
```

Let's use this other image because
1. it has builds for both amd64 and arm64 platforms, so it will be faster for both of us
2. it has luarocks already installed, so we won't need to do the apt-get stuff, which is slow

We'll need to use a a JSON library that doesn't need build tools (gcc, cmake, etc) installed (so don't need to use `apt-get`). [`dkjson`](https://luarocks.org/modules/dhkolf/dkjson) looks like a good one.